test/runner/browserstack/api.js

```diff
@@ -148,17 +158,54 @@ export async function filterBrowsers( filter ) {
 	const filterOsVersion = ( filter.os_version ?? "" ).toLowerCase();
 	const filterDevice = ( filter.device ?? "" ).toLowerCase();
 
-	return browsers.filter( ( browser ) => {
+	const filteredWithoutVersion = browsers.filter( ( browser ) => {
 		return (
 			( !filterBrowser || filterBrowser === browser.browser.toLowerCase() ) &&
-			( !filterVersion ||
-				matchVersion( browser.browser_version, filterVersion ) ) &&
 			( !filterOs || filterOs === browser.os.toLowerCase() ) &&
-			( !filterOsVersion ||
-				filterOsVersion === browser.os_version.toLowerCase() ) &&
+			( !filterOsVersion || matchVersion( browser.os_version, filterOsVersion ) ) &&
 			( !filterDevice || filterDevice === ( browser.device || "" ).toLowerCase() )
 		);
 	} );
+
+	if ( !filterVersion ) {
+		return filteredWithoutVersion;
+	}
+
+	if ( filterVersion.startsWith( "latest" ) ) {
+		const groupedByName = filteredWithoutVersion
+			.filter( ( b ) => rfinalVersion.test( b.browser_version ) )
+			.reduce( ( acc, browser ) => {
+				acc[ browser.browser ] = acc[ browser.browser ] || [];
+				acc[ browser.browser ].push( browser );
+				return acc;
+			}, {} );
+
+		const filtered = [];
+		for ( const group of Object.values( groupedByName ) ) {
+			const num = rlatest.exec( filterVersion );
+			const latest = group[ group.length - 1 ];
+
+			if ( !latest.browser_version ) {
+				filtered.push( latest );
+				continue;
+			}
```

Comment:
Can you elaborate this block?

Reply:
I'll add a comment. Mobile devices do not have browser version, so it skips the version check, but includes the latest in the list if it made it through filtering. That way, the latest of all browsers is shown. But, `latest-1` doesn't work as there's no version to compare to.

Reply:
For mobile browsers, can't we we use the `lastest` logic to compare OS versions?

Reply:
I talk about that in the PR description. It's not bad for iOS, but OS versions do not even always contain numbers. Also, I didn't think it worth the extra effort given that OS versions don't change nearly as often.

Reply:
OK, makes sense.